### DISCUSSION TITLE
How to fix the menu at a distance of 100px

### DISCUSSION THREAD

**Discussion by ToitenAn** [community/community#103242] [Accessibility] [answered]
### Select Topic Area

Question

### Body

How to fix the menu at a distance of 100px from the top of the page, if it's more than 100px, it will be hidden( html,css and JS) pls

**Comment by bug3** [comment]
Hello @ToitenAn  Can you share the image or codes of the menu?

**Comment by amankr1469** [comment]
Define the styles for your menu and content. For the menu, set it to have a fixed position, 100px from the top, with a background color, padding, and z-index to ensure it's always visible. Also, add styles for the menu items (like links). For the content, make sure it has enough padding at the top to prevent it from being covered by the menu. 
Write a JavaScript function that listens for the scroll event on the window. When the user scrolls, check the current vertical scroll position. If it's greater than 100px, change the menu's top position to 0px, effectively sticking it to the top of the page. If the scroll position is less than or equal to 100px, set the menu's top position back to 100px, making it visible at its original position.

**Comment by gil-son** [comment]
Create a simple HTML structure for your page with a navigation menu:

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="styles.css">
    <title>Fixed Menu Example</title>
</head>
<body>
    <header>
        <nav>
            <ul>
                <li><a href="#">Home</a></li>
                <li><a href="#">About</a></li>
                <li><a href="#">Services</a></li>
            </ul>
        </nav>
    </header>
    <div class="content">
        <h1>Your Page Content</h1>
    </div>
    <script src="script.js"></script>
</body>
</html>
```
Create a CSS file (styles.css) to style your navigation menu and set initial properties:

```
body {
    margin: 0;
    padding: 0;
}

header {
    background-color: #333;
    color: #fff;
    padding: 10px;
    position: fixed;
    width: 100%;
    top: -100px; /* Initially hide the header above the viewport */
    transition: top 0.3s; /* Add smooth transition */
}

nav ul {
    list-style: none;
    margin: 0;
    padding: 0;
    display: flex;
}

nav li {
    margin-right: 20px;
}

nav a {
    text-decoration: none;
    color: #fff;
    font-weight: bold;
    font-size: 18px;
}

.content {
    margin-top: 150px; /* Add margin to prevent content overlap with fixed header */
}

```

Create a JavaScript file (script.js) to handle the scroll event and toggle the visibility of the fixed menu:

```
document.addEventListener("DOMContentLoaded", function () {
    var header = document.querySelector("header");

    window.addEventListener("scroll", function () {
        if (window.scrollY > 100) {
            header.style.top = "0";
        } else {
            header.style.top = "-100px";
        }
    });
});

```

**Comment by UdaySMukherjee** [comment, marked as answer]
HTML (index.html):
Create a new file named index.html and copy the HTML content into it.


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="styles.css">
    <title>Fixed Menu Example</title>
</head>
<body>
    <header id="fixed-header">
        
        <nav>
            <ul>
                <li><a href="#">Home</a></li>
                <li><a href="#">About</a></li>
                <li><a href="#">Contact</a></li>
            </ul>
        </nav>
    </header>

    
    <div id="content">
        
    </div>

    <script src="script.js"></script>
</body>
</html>
CSS (styles.css):
Create a new file named styles.css and copy the CSS content into it.

/* styles.css */
body {
    margin: 0;
    padding: 0;
}

#fixed-header {
    position: fixed;
    top: -100px; /* Initially hide the menu above the viewport */
    left: 0;
    width: 100%;
    background-color: #333; /* Choose your background color */
    padding: 10px 0; /* Adjust padding according to your design */
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1); /* Optional: Add a subtle shadow */
    transition: top 0.3s; /* Smooth transition effect */
}

#fixed-header nav ul {
    list-style: none;
    margin: 0;
    padding: 0;
    text-align: center;
}

#fixed-header nav ul li {
    display: inline-block;
    margin: 0 10px;
}

#fixed-header nav ul li a {
    text-decoration: none;
    color: white;
    font-weight: bold;
    font-size: 16px;
}
JavaScript (script.js):
Create a new file named script.js and copy the JavaScript content into it.

// script.js
document.addEventListener("scroll", function() {
    var header = document.getElementById("fixed-header");

    // Check if the scroll position is beyond 100px
    if (window.scrollY > 100) {
        header.style.top = "0"; // Show the fixed menu
    } else {
        header.style.top = "-100px"; // Hide the fixed menu
    }
});

**Comment by cehfisher** [comment]
Hi @ToitenAn - this issue appears to be resolved [per this comment](https://github.com/orgs/community/discussions/103242#discussioncomment-8355320). Could you please confirm and close out this issue? Thank you! (cc @queenofcorgis)
